packages/block-editor/src/components/rich-text/use-paste-handler.js

```diff
@@ -58,10 +58,18 @@ export function usePasteHandler( props ) {
 			const isInternal =
 				event.clipboardData.getData( 'rich-text' ) === 'true';
 
+			const trimmedPlainText = plainText.trim();
+
+			const isPasteableUrl =
+				__unstableEmbedURLOnPaste &&
+				isURL( trimmedPlainText ) &&
+				// For the link pasting feature, allow only http(s) protocols.
+				/^https?:/.test( trimmedPlainText );
+
 			// If the data comes from a rich text instance, we can directly use it
 			// without filtering the data. The filters are only meant for externally
 			// pasted content and remove inline styles.
-			if ( isInternal ) {
```

Comment:
This works, but I would much rather make the logic below that handles `__unstablePasteRule` transforms reusable and run it here as well, instead of making an exception for URLs and letting that from through the paste handler for nothing. What do you think?

Reply:
Here's what I meant: [Rich text: fix link paste for internal paste](https://github.com/WordPress/gutenberg/pull/58786#top)